IOS additions

timidri opened the pull request:
- Added IOS profile and hiera data
- Fixed path for hiera keys for other devices

timidri commented:
@murdok5 Thanks, updated the class name. 
About demo using half hiera half code, it's a good point. The pro is indeed that the user sees how puppet code works without an additional complexityl layer. The con is that we actually don't necessarily the customer "best practices".

What would be a good trade-off is to add 'hard-wired' configuration into the profile which needs to be equal for all devices classified with that profile ('static' items). Typically syslog and DNS are _not_ static unless the environment is really simple, but we can pretend it is for this use case.

Let's do the hard-wired banner :-)

murdok5 commented:
@timidri I have always seen the demo env as a balance of the possible with reality, where best practices come later at implementation time.  Mostly because best practices dont demo well, they scale well.  But either way I think it would benefit from some static items and some dynamic items (which I would say to basically ANY demo profile).

timidri commented:
@murdok5 Added DNS and syslog management to puppet code.
Also, added Puppetfile pre-requisites to actually use the cisco_ios module.
Also, added `profile::infrastructure::network::device_manager` to automate configration of puppet device.
Finally, added the test Catalyst 2960 creds to hieradata. Note: we need to think about this, since if everyone starts demoing this at the same time, chaos will ensue. Any thoughts on this?

timidri commented:
@murdok5 Can this be merged?
